app/src/main/java/com/example/androidtrainingproject/ui/products/ProductDetailsScreen.kt

```diff
@@ -17,22 +17,24 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
-import androidx.navigation.NavController
 import com.example.androidtrainingproject.R
 import com.example.androidtrainingproject.ui.shared.AppTopBar
 import com.example.androidtrainingproject.ui.shared.WideButton
 import com.example.androidtrainingproject.ui.theme.LabelText
+import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
 
+@Destination
 @Composable
-fun ProductDetails(navController: NavController, productId: Int) {
+fun ProductDetails(destinationsNavigator: DestinationsNavigator, productId: Number) {
```

Comment:
You can pass the `productId` directly to the view model without the view even knowing about it using `SavedStateHandle` demonstrated on the last lecture.